wireapp/kalium
# feat: Group Conversation - Receipt Mode System Message

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

[AR-2712 - Show a system message when read receipts are turned on](https://wearezeta.atlassian.net/browse/AR-2712)

This PR is only for handling receipt mode when a new group event is received or in-app user creates a group.
The handling of receipt mode from changes in group settings screen will be handled in a later PR.

### Issues

New group conversations [from event or user in-app created] were not adding a system message for receipt mode.

### Causes (Optional)

It was not implemented.

### Solutions

Add implementation to always add a system message informing the members of the conversation what is the receipt mode status if the creator/receiver of the new group event matches the criteria of:
- isSelfATeamMember
- conversation is of GROUP type
- receipt mode status is NOT NULL (for users without a team it will be null when invoking `CreateGroupConversationUseCase` thus not showing for them)

### Dependencies (Optional)

A PR in AR will be raised upon this one being merged


### Testing

#### How to Test

Currently the only way to test this in Kalium is to build this branch in AR and create/receive a new group event and then check `MessageNewConversationReceiptModeContent` table values that is contains the selected receipt mode.
